.github/workflows/ci-sgx-all.yml

```diff
@@ -0,0 +1,54 @@
+name: CI - SGX ALL
+
+on:
+  workflow_call:
+  push:
+    branches: ["main"]
+    paths:
+      - "provers/sgx/**"
+  pull_request:
+    paths:
+      - "provers/sgx/**"
+
+jobs:
+    build-test-sgx:
+      name: Build and test sgx with Docker
+      uses: ./.github/workflows/ci-build-test-reusable.yml
+      with:
+        version_name: "sgx"
+        version_toolchain: "stable"
+
+    build-test-sgx-docker:
+      name: Build and test sgx with Docker
+      uses: ./.github/workflows/ci-sgx-docker.yml
+        
+    build-test-sgx-hardware:
+      name: Build and test sgx in hardware
```

Comment:
can we follow the convention here and call another workflow, `ci-sgx-hardware.yml`? i thought `ci-sgx-native.yml` could be good, but seems like we already have `ci-native.yml`.